graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/Elasticsearch7Module.java

```diff
@@ -42,19 +42,25 @@
 import org.graylog2.storage.SearchVersion;
 
 public class Elasticsearch7Module extends VersionAwareModule {
-
     private final SearchVersion supportedVersion;
+    private final boolean useComposableIndexTemplates;
 
-    public Elasticsearch7Module(final SearchVersion supportedVersion) {
+    public Elasticsearch7Module(final SearchVersion supportedVersion, boolean useComposableIndexTemplates) {
         this.supportedVersion = supportedVersion;
+        this.useComposableIndexTemplates = useComposableIndexTemplates;
     }
 
     @Override
     protected void configure() {
         bindForSupportedVersion(StreamsForFieldRetriever.class).to(StreamsForFieldRetrieverES7.class);
         bindForSupportedVersion(CountsAdapter.class).to(CountsAdapterES7.class);
         bindForSupportedVersion(ClusterAdapter.class).to(ClusterAdapterES7.class);
-        bindForSupportedVersion(IndicesAdapter.class).to(IndicesAdapterES7.class);
+        if (useComposableIndexTemplates) {
+            bindForSupportedVersion(IndicesAdapter.class).to(IndicesAdapterES7.class);
+        } else {
+            bindForSupportedVersion(IndicesAdapter.class).to(LegacyIndicesAdapterES7.class);
+        }
```

Comment:
There is A LOT of code duplication in `IndicesAdapterES7` and `LegacyIndicesAdapterES7`.
They are different only when it comes to templates.
What about injecting the feature flag directly to the adapter and keeping just single `IndicesAdapterES7` class?
(You could consider moving only template-related code outside of `IndicesAdapterES7` class later on, and spreading it to 2 different classes).

PS
It is the same in OS version.

Reply:
I considered that too, but I think this way it will be easier to just get rid of `LegacyIndicesAdapterES7` when we remove the feature flag again.

Reply:
I'd prefer smth like that:
```
public class IndicesAdapterES7 implements IndicesAdapter {
   
    (...)

    private TemplateHandler templateHandler;

    @Inject
    public IndicesAdapterES7(ElasticsearchClient client,
                             StatsApi statsApi,
                             CatApi catApi,
                             ClusterStateApi clusterStateApi,
                             TemplateHandler templateHandler) {
        this.client = client;
        this.statsApi = statsApi;
        this.catApi = catApi;
        this.clusterStateApi = clusterStateApi;
        this.templateHandler = templateHandler;
    }

    (...)
```
And binding decision for `TemplateHandler`, which, without looking into details of the code, should be as easily removable as current solution, but with less code duplication.

Reply:
Good idea! Did that in [b39f309](https://github.com/Graylog2/graylog2-server/pull/15948/commits/b39f30939e0229d9b9e7e587943b5cad84b77829).

Reply:
Thanks!